src/rgw/driver/rados/rgw_rest_user.cc

```diff
@@ -551,11 +569,20 @@ void RGWOp_Subuser_Modify::execute(optional_yield y)
   rgw_user uid(uid_str);
 
   RESTArgs::get_string(s, "subuser", subuser, &subuser);
-  RESTArgs::get_string(s, "secret-key", secret_key, &secret_key);
+  RESTArgs::get_string(s, "secret", secret_key, &secret_key);
   RESTArgs::get_string(s, "access", perm_str, &perm_str);
   RESTArgs::get_string(s, "key-type", key_type_str, &key_type_str);
   RESTArgs::get_bool(s, "generate-secret", false, &gen_secret);
 
+  // if keys cap is not set to write a keys cannot be generated or set
+  if(gen_secret && (s->user->get_info().caps.check_cap("keys", RGW_CAP_WRITE) == -EPERM)) {
+    op_ret = -EPERM;
+    return;
+  } else if((!secret_key.empty()) && (s->user->get_info().caps.check_cap("keys", RGW_CAP_WRITE) == -EPERM)) {
```

Comment:
similar comment about not checking for `-EPERM` specifically. you can also consider combining these two with something like:
```c++
  // require "keys=write" cap to generate or modify secret_key
  if (gen_secret || !secret_key.empty()) {
    op_ret = s->user->get_info().caps.check_cap("keys", RGW_CAP_WRITE);
    if (op_ret < 0) {
      return;
    }
  }
```